src/packagedcode/pypi.py

```diff
@@ -832,6 +842,310 @@ def parse(cls, location, package_only=False):
+        packages = toml_data.get('package')
+        if not packages:
+            return
+
+        version = toml_data.get('version')
+        requires_python = toml_data.get('requires-python')
+
+        dependencies = []
+        for package in packages:
+            dependencies_for_resolved = []
+
+            # Handle dependencies - UV uses a different format than Poetry
+            deps = package.get("dependencies") or []
+            for dep in deps:
+                if isinstance(dep, dict):
+                    # UV format: {name: "package-name", marker: "condition"}
+                    dep_name = dep.get('name')
+                    marker = dep.get('marker')
+                    purl = PackageURL(
+                        type=cls.default_package_type,
+                        name=dep_name,
+                    )
+                    dependency = models.DependentPackage(
+                        purl=purl.to_string(),
+                        extracted_requirement=marker,
+                        scope="dependencies",
+                        is_runtime=True,
+                        is_optional=False,
+                        is_direct=True,
+                        is_pinned=False,
+                    )
+                    dependencies_for_resolved.append(dependency.to_dict())
+                elif isinstance(dep, str):
+                    # Simple string dependency
+                    dependency = parse_dependency_requirement(
+                        requirement=dep,
+                        scope='dependencies',
+                        is_runtime=True,
+                    )
+                    if dependency:
+                        dependencies_for_resolved.append(dependency.to_dict())
+
+            name = package.get('name')
+            version = package.get('version')
+            urls = get_pypi_urls(name, version)
+            
+            package_data = dict(
+                datasource_id=cls.datasource_id,
+                type=cls.default_package_type,
+                primary_language='Python',
+                name=name,
+                version=version,
+                is_virtual=True,
+                dependencies=dependencies_for_resolved,
+                **urls,
+            )
+            resolved_package = models.PackageData.from_data(package_data, package_only)
+
+            dependency = models.DependentPackage(
+                purl=resolved_package.purl,
+                extracted_requirement=None,
+                scope=None,
+                is_runtime=True,
+                is_optional=False,
+                is_direct=False,
+                is_pinned=True,
+                resolved_package=resolved_package.to_dict()
+            )
+            dependencies.append(dependency.to_dict())
+
+        extra_data = {}
+        extra_data['python_version'] = requires_python
+        extra_data['lock_version'] = version
+
+        package_data = dict(
+            datasource_id=cls.datasource_id,
```

Comment:
there are a lot more package data fields which can be parsed and added, please modify to include those. See https://github.com/aboutcode-org/scancode-toolkit/blob/c05a13d87b41cfc629ad4497cf40bcb9c26ee703/src/packagedcode/models.py#L548 for more details for all the fields and refer to other package data parsers too